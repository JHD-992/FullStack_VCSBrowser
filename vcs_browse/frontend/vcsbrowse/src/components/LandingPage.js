import React from 'react'

function LandingPage (props) {

    return <div>
    <link rel="stylesheet" href="/stylesheets/index.css"/>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
    <title>VCS Browser</title>
    <h1>VCS Browser</h1>
    <p>
      Welcome to the VCS browser project. This project will endevour to provide easrch functionality across 3 different VCS systems GitHub, GitLab
      and BitBucket.
    </p>
    <div class="topnav">
      <a class="active" href="/">Home</a>
      <div class="search-container">
        <form action="/gitLabUsers">
          <input type="text" placeholder="Search on GitLab" name="search" />
          <button type="submit"><i class="fa fa-search"></i></button>
        </form>
      </div>
      <div class="search-container">
        <form action="/bitBucketUsers">
          <input type="text" placeholder="Search on BitBucket" name="search"/>
          <button type="submit"><i class="fa fa-search"></i></button>
        </form>
      </div>
      <div class="search-container">
        <form action="/gitHubUsers">
          <input type="text" placeholder="Search on GitHub" name="search"/>
          <button type="submit"><i class="fa fa-search"></i></button>
        </form>
      </div>
      </div>
      </div>
    }

export default LandingPage