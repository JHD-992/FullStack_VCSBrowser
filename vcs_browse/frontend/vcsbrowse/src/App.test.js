import React from 'react';
import LandingPage from '../src/components/LandingPage';
import renderer from 'react-test-renderer';

test ('renders correctly', () => {
    const tree = renderer
    .create(<LandingPage />)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

const fetch = jest.fn()

it("fetch api to interact with other api's", async () => {
  fetch.mockReturnValueOnce(JSON.stringify({user: { id: "jaco" }}));

  const call = await fetch();
  const data = JSON.parse(call);
  const word = data.user.id;

  expect(word).toEqual("jaco");
  expect(fetch).toHaveBeenCalledTimes(1);
})