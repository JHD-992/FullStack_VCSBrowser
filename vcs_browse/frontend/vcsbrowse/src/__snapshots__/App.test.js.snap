// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <link
    href="/stylesheets/index.css"
    rel="stylesheet"
  />
  <link
    href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
    rel="stylesheet"
  />
  <title>
    VCS Browser
  </title>
  <h1>
    VCS Browser
  </h1>
  <p>
    Welcome to the VCS browser project. This project will endevour to provide easrch functionality across 3 different VCS systems GitHub, GitLab and BitBucket.
  </p>
  <div
    class="topnav"
  >
    <a
      class="active"
      href="/"
    >
      Home
    </a>
    <div
      class="search-container"
    >
      <form
        action="/gitLabUsers"
      >
        <input
          name="search"
          placeholder="Search on GitLab"
          type="text"
        />
        <button
          type="submit"
        >
          <i
            class="fa fa-search"
          />
        </button>
      </form>
    </div>
    <div
      class="search-container"
    >
      <form
        action="/bitBucketUsers"
      >
        <input
          name="search"
          placeholder="Search on BitBucket"
          type="text"
        />
        <button
          type="submit"
        >
          <i
            class="fa fa-search"
          />
        </button>
      </form>
    </div>
    <div
      class="search-container"
    >
      <form
        action="/gitHubUsers"
      >
        <input
          name="search"
          placeholder="Search on GitHub"
          type="text"
        />
        <button
          type="submit"
        >
          <i
            class="fa fa-search"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;
